### DISCUSSION TITLE
SwinUNETR training details on MSD dataset

### DISCUSSION THREAD

**Discussion by fanlimua** [Project-MONAI/MONAI#7064] [Q&A]
I'm attempting to replicate the results presented in your paper titled "Self-Supervised Pre-Training of Swin Transformers for 3D Medical Image Analysis" using the MSD lung and hippocampus datasets in Appendix, Section B.2. 

I followed [spleen segmentation](https://github.dev/Project-MONAI/tutorials/blob/87792475f4476dcc63f92e47e109dde18cdfb30d/3d_segmentation/spleen_segmentation_3d.ipynb) tutorial to preprocess data and generate 3D augmentations, and I've aligned all the parameters in the transformers with your paper.  But I encountered several issues during the training:

1. Could you please provide details on how you preprocessed the validation and test data?
2. Did you assess the memory requirements for the sliding_window_inference operation during validation? I tried to modify the roi_size from (160,160,160,) to (64,64,64) (batch_size and num_works are 1),  and encountered Out Of Memory (OOM) errors.
3. Your paper states that "each hippocampus MRI image is interpolated by voxel spacing of 0.2x0.2x0.2, then applied spatial padding to 96x 96x96 as the input size of Swin UNETR model". But if the input size of hippocampus is (32,51,31),  it would be converted to (156,251,151) after applying the Spacingd transformer. Spatial padding operation cannot covert (156,251,151) to (96,96,96).  Could you clarify how you handled this issue?
4. What is the loss function for fine-tuning on MSD dataset?
5. How many epochs did you train on the lung and hippocampus datasets?
	
If it's possible, could you please share the source code for data preprocessing and training on lung and hippocampus dataset? I really appreciate that.

**Comment by KumoLiu** [comment]
Hi @fanlimua, I think you could refer to this [tutorial](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/swin_unetr_brats21_segmentation_3d.ipynb).
There is also another tutorial about Swin UNETR. https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/swin_unetr_btcv_segmentation_3d.ipynb

Hope it helps, thanks!
